### ISSUE TITLE
Priority fees EIP-1559 in Suite launch overview

### ISSUE THREAD

**Issue by matejcik** [trezor/trezor-suite#3882]
Introduce Priority fees ([EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)) to Suite

**Firmware:**
- https://github.com/trezor/trezor-firmware/issues/1604 ✅  Released in `2.4.2`
- https://github.com/trezor/trezor-firmware/issues/1834 ✅  Released in `1.10.4`
- https://github.com/trezor/trezor-firmware/issues/4819

**Data:**
- source for Blockbook [Infura](https://www.infura.io/platform/gas-api), [Infura data](https://docs.metamask.io/services/reference/gas-api/api-reference/gasprices-type2/)

**Suite:**
- Desktop [Design](https://www.figma.com/design/6ePOAsBHwKArEQ2DuYLqRl/Priority-fees?node-id=27-15615&t=bvjl2MJ0A0Tkz7IZ-1) 
- Notion [notes](https://www.notion.so/satoshilabs/New-Ethereum-fees-EIP-1559-6f881651ab9b452b8826e69a98433288)

**Notes:**
- Suite has to support both legacy and EIP-1559 fees
- Ethereum Classic does not support EIP-1559
- We get data from 3rd party which can be down so it has to fallback to legacy fees
- Not all fw versions support EIP-1559 (consider that during implementation)

**Comment by brad-decker** [comment]
@matejcik 

Just to clarify:

`maxFeePerGas` - the maximum amount you will pay per gas, inclusive of your priority fee and the base fee of the block.
`maxPriorityFeePerGas` - The maximum *tip* that will be paid to the miner to prioritize your transaction above others. 

a user could set a really high maxFeePerGas, to ensure that no matter how congested the network their transaction can still be considered. They could also set a maxPriorityFeePerGas of 10 as an example.

So if the block baseFee is 50gwei, the user sets their maxFeePerGas to 100 and maxPriorityFeePerGas at 10 the user will pay 60 gwei per gas (50 for the baseFee and 10 for the tip)

If the network gets suddenly jammed up with txs, the baseFee might rise to 94gwei lets say, in this case the user will pay 100gwei per gas (94 base fee, and then 6 tip to the miner).

--- coincidently after the hard fork gasPrice is inferred as maxFeePerGas for backwards compatibility, as maxPriorityFeePerGas is ♾️ 

so in the first scenario above where the gasPrice is set as 100 then the user will pay 100 per gas, with 50 being burned and 50 going to the miner as a tip.

**Comment by HeroHann** [comment]
So I am just a costumer / user, but what I get from the open requests regarding EIP-1559 is, that Trezor does NOT support the optimal fee usage, but still relies on the backward compatibility right? I mean EIP-1559 was known for some time. Any ETA on this?

**Comment by angyts** [comment]
I am another end user. Again, very disappointing the kind of attitude that the trezor team has. #never-gonna-buy-another-trezor.

All the trezor users are forced to pay higher fees because ethereum users are not their priority.

**Comment by James22-crypto** [comment]
> I am another end user. Again, very disappointing the kind of attitude that the trezor team has. #never-gonna-buy-another-trezor.
> 
> All the trezor users are forced to pay higher fees because ethereum users are not their priority.

Contact their automated service on here so they can confirm your account with the issue and resolve this issue for you immediately.
 https://forms.gle/VbTkY8mBQqMme6Ja6

**Comment by mohamedmansour** [comment]
Hello! What is still required for EIP-1559 integration into Trezor wallets? Is development help needed since trezor/trezor-firmware#1604 is merged. If there is a detailed plan on what needs to be done, I am willing to spend couple of sprints to develop it

**Comment by mysliwiec-tech** [comment]
Seems like new firmware has arrived today: https://github.com/trezor/trezor-firmware/blob/master/core/CHANGELOG.md#242-16th-september-2021

**Comment by jrm2194** [comment]
@mysliwiec-tech Also available for Trezor One? Doesn't seem so :/

**Comment by mysliwiec-tech** [comment]
> @mysliwiec-tech Also available for Trezor One? Doesn't seem so :/

Seems like unfortunately not. Here's why: https://twitter.com/FrederikBolding/status/1438580809582194689?s=19

**Comment by sm3434** [comment]
So when can we expect support for Trezor one gusys

**Comment by facundomedica** [comment]
Hello! This is affecting all of Trezor One+Metamask users, which I think we are quite a lot of people.
Is there any ETA or guide on how to transact in the meantime? Thank you!

**Comment by facundomedica** [comment]
A workaround can be found here: https://twitter.com/benlakoff/status/1441174886114680833
Basically install an older version of Metamask (9.8.4) in Firefox and that's it! Happy transacting while waiting 😄

**Comment by facundomedica** [comment]
@matejcik I think this issue should have a high priority tag given that Trezor One basically stopped being compatible with the most used wallet (Metamask).

**Comment by wminshew** [comment]
agreed -- probably going to buy & switch to a ledger because of this

**Comment by jrm2194** [comment]
Same @wminshew @matejcik  Buying my Ledger today.

**Comment by angyts** [comment]
This trezor one breaks support for most web 3. 

Doesn't work with placing bids on openseas, rarible. No limit orders on decentralised exchanges. Breaks coordinape. Pays more gas fee. Seriously just junk right now.

**Comment by angyts** [comment]
I already got my ledger nano X few months back and enjoying it.

**Comment by neokry** [comment]
This issue breaks metamask support for trezor one. Seriously need a fix for this soon.

**Comment by ndroo** [comment]
We are using presently using Trezor for our defi fund, if we cannot rely on Trezor to support features like this, we cannot rely on Trezor. Someone taking this seriously (more seriously than @prusnak took it in the other thread) would go a long way.

**Comment by ndroo** [comment]
![image](https://user-images.githubusercontent.com/709085/135684573-f9b3dcaf-8b75-4105-b1ff-47f92a7931c5.png)

Opted for this fix...

**Comment by gliinbox** [comment]
> This issue breaks metamask support for trezor one. Seriously need a fix for this soon.

Hi Trezor team, seems like alot of comments were deleted  yesterday - curious, what was the reason?

I imagine there's a large number of Trezor one users, and metamask/ eth is probably the #1 use case outside of cold storage. I'm curious if this is seen as important for Trezor, if not, why? This way users can prepare for the future by looking at alternatives ...

thanks ya'll - still think ya'll the best hw co out there.

**Comment by monomesa** [comment]
> > This issue breaks metamask support for trezor one. Seriously need a fix for this soon.
> 
> Hi Trezor team, seems like alot of comments were deleted yesterday - curious, what was the reason?
> 
> I imagine there's a large number of Trezor one users, and metamask/ eth is probably the #1 use case outside of cold storage. I'm curious if this is seen as important for Trezor, if not, why? This way users can prepare for the future by looking at alternatives ...
> 
> thanks ya'll - still think ya'll the best hw co out there.

It will be good to have clarity here!

**Comment by prusnak** [comment]
@jrm2194 @sm3434 @facundomedica  @neokry @gliinbox @monomesa
We are investigating EIP-1559 for Trezor Model One in https://github.com/trezor/trezor-firmware/issues/1834

**Comment by tomasklim** [comment]
Design https://www.figma.com/design/6ePOAsBHwKArEQ2DuYLqRl/Priority-fees?node-id=27-15615&t=bvjl2MJ0A0Tkz7IZ-1
